src/tests/test_connection.py

```diff
@@ -45,6 +46,86 @@ def test_get_curr_connection_info(mock_postgres):
     }
 
 
+def test_get_curr_sqlglot_dialect_no_curr_connection(monkeypatch):
+    monkeypatch.setattr(Connection, "_get_curr_connection_info", lambda: None)
+    assert Connection._get_curr_sqlglot_dialect() is None
+
+
+@pytest.mark.parametrize(
+    "sqlalchemy_connection_info, expected_sqlglot_dialect",
+    [
+        (
+            {
+                "dialect": "duckdb",
+                "driver": "duckdb_engine",
+                "server_version_info": [8, 0],
+            },
+            "duckdb",
+        ),
+        (
+            {
+                "dialect": "mysql",
+                "driver": "pymysql",
+                "server_version_info": [10, 10, 3, 10, 3],
+            },
+            "mysql",
+        ),
+        # sqlalchemy and sqlglot have different dialect name, test the mapping dict
+        (
+            {
+                "dialect": "sqlalchemy_mock_dialect_name",
+                "driver": "sqlalchemy_mock_driver_name",
+                "server_version_info": [0],
+            },
+            "sqlglot_mock_dialect",
+        ),
+        (
+            {
+                "dialect": "only_support_in_sqlalchemy_dialect",
+                "driver": "sqlalchemy_mock_driver_name",
+                "server_version_info": [0],
+            },
+            "only_support_in_sqlalchemy_dialect",
+        ),
+    ],
+)
+def test_get_curr_sqlglot_dialect(
+    monkeypatch, sqlalchemy_connection_info, expected_sqlglot_dialect
+):
+    monkeypatch.setattr(
+        Connection, "_get_curr_connection_info", lambda: sqlalchemy_connection_info
+    )
+    monkeypatch.setattr(
```

Comment:
Do you mean this?

```
def test_get_curr_sqlglot_dialect(
    monkeypatch, sqlalchemy_connection_info, expected_sqlglot_dialect
):
    mock_connection = Mock()
    mock_connection._get_curr_connection_info = lambda: sqlalchemy_connection_info
    mock_connection.DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING = {"sqlalchemy_mock_dialect_name": "sqlglot_mock_dialect"}
    assert mock_connection._get_curr_sqlglot_dialect() == expected_sqlglot_dialect
```

Got some questions:
If we mock the original `sql.connection`, how we test the `_get_curr_sqlglot_dialect` function inside since it's parent class is being mocked?